rest/config.go

```diff
@@ -1430,10 +1438,38 @@ func (sc *ServerContext) fetchDatabase(ctx context.Context, dbName string) (foun
 			cnf.KeyPath = sc.Config.Bootstrap.X509KeyPath
 		}
 		base.TracefCtx(ctx, base.KeyConfig, "Got config for bucket %q with cas %d", bucket, cas)
-		return true, &cnf, nil
+		return true, nil
 	}
 
-	return false, nil, nil
+	err = sc.foreachDbConfig(callback)
+	if err != nil {
+		return false, nil, nil
```

Comment:
Should we be returning the error here?

```suggestion
		return false, nil, err
```